src/libcore/ops/function.rs

```diff
@@ -61,29 +72,36 @@ pub trait Fn<Args> : FnMut<Args> {
     extern "rust-call" fn call(&self, args: Args) -> Self::Output;
 }
 
-/// A version of the call operator that takes a mutable receiver.
+/// The version of the call operator that takes a mutable receiver.
+///
+/// Instances of `FnMut` can be called repeatedly and may mutate state.
 ///
-/// Closures that might mutably reference captured variables automatically
-/// implement this trait, which allows them to be invoked. For immutably
-/// referenced captures, see [`Fn`], and for consuming the captures, see
-/// [`FnOnce`].
+/// `FnMut` is implemented automatically by closures which take mutable
+/// references to captured variables, as well as all types that implement
+/// [`Fn`], e.g. (safe) [function pointers][] (since `FnMut` is a supertrait of
+/// [`Fn`]). Additionally, for any type `F` that implements `FnMut`, both `&F`
+/// and `&mut F` implement `FnMut`, too.
 ///
-/// You can use the [`Fn`] traits when you want to accept a closure as a
-/// parameter. Since [`FnOnce`] is a supertrait of `FnMut`, any instance of
-/// `FnMut` can be used where a [`FnOnce`] is expected, and since [`Fn`] is a
-/// subtrait of `FnMut`, any instance of [`Fn`] can be used where `FnMut` is
-/// expected.
+/// Since [`FnOnce`] is a supertrait of `FnMut`, any instance of `FnMut` can be
+/// used where a [`FnOnce`] is expected, and since [`Fn`] is a subtrait of
+/// `FnMut`, any instance of [`Fn`] can be used where `FnMut` is expected.
+///
+/// Use `FnMut` as a bound when you want to accept a parameter of function-like
+/// type and need to call it repeatedly, without requiring it not to mutate
+/// state. If you require the parameter to not mutate state, use [`Fn`] as a
```

Comment:
I'd prefer to say "while allowing it to mutate state", just to get around the double-negative there

Ditto for "If you don't want the parameter to mutate state,"